app/controllers/application_controller.rb

```diff
@@ -12,15 +12,20 @@ def authenticate!
   end
 
   def authorize!
-    allowed = AuthorizationService.call(
+    auth_result = AuthorizationService.call(
       @current_client_role,
       controller_name,
       action_name,
       params,
       authorization_object,
     )
 
-    head(:forbidden) unless allowed
+    case auth_result
+    when :forbidden
+      head(:forbidden)
+    when :no_op
+      head(:no_content)
```

Comment:
I chose a 204 No Content over a 304 Not Modified because 304 is very commonly used and, despite the more appropriate-seeming title, has its own quite specific meaning that is supposed only to apply to HEAD and GET requests, so overriding its meaning felt like it could get confusing.